src/execute/ExecuteView.tsx

```diff
@@ -164,116 +83,57 @@ export const ExecuteView: React.FC<Props> = ({
     });
   };
 
-  if (isLoading || !exerciseXML) {
-    return <SkeletonScreen />;
-  }
-
-  const blogArticle = new UIBlogArticle(exerciseXML || "");
-  console.log(blogArticle._xml);
+  const renderMode = () => {
```

Comment:
why is this a function instead of component?

Reply:
Initially used it as function as all the props to the modes are local to the ExecuteView, Created a component now and sent the required fields as props.